flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -467,6 +487,128 @@ public void registerAtResourceManager(final String address) {
 		//TODO:: register at the RM
 	}
 
+	@RpcMethod
+	public NextInputSplit requestNextInputSplit(final JobVertexID vertexID, final ExecutionAttemptID executionAttempt) {
+		final byte[] serializedInputSplit;
+
+		final Execution execution = executionGraph.getRegisteredExecutions().get(executionAttempt);
+		if (execution == null) {
+			log.error("Can not find Execution for attempt {}.", executionAttempt);
+			return null;
```

Comment:
Let's let this throw an exception - after all, the caller receiver should react to the fact that the JobManager is not aware of its execution any more.